modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/api/ItCommonApiTest.java

```diff
@@ -120,6 +122,27 @@ public void testSessionExpiration() throws Exception {
         ses2.execute(null, "SELECT 2 + 2").close();
     }
 
+    /** Check correctness of planning timeout. */
+    @Test
+    public void testPlanningTimeout() {
+        IgniteSql sql = igniteSql();
+
+        sql("CREATE TABLE TST1(id INTEGER PRIMARY KEY, val INTEGER)");
+
+        Long oldPlannerTimeout = (Long) IgniteTestUtils.getFieldValue(queryProcessor(), SqlQueryProcessor.class, "plannerTimeout");
```

Comment:
Can we introduce 'plannerTimeout' option and pass it e.g. via session? 
Or use system property instead of reflection hack.